lib/train-winrm/transport.rb

```diff
@@ -164,32 +165,41 @@ def load_needed_dependencies!
         spec_version = WINRM_FS_SPEC_VERSION.dup
         logger.debug("winrm-fs requested," \
           " loading WinRM::FS gem (#{spec_version})")
-        gem "winrm-fs", spec_version
-        first_load = require "winrm-fs"
-        load_winrm_transport!
+        load_dependency("winrm-fs", spec_version, "Could not load or activate WinRM::FS")
+
+        spec_version = WINRM_ELEVATED_SPEC_VERSION.dup
+        logger.debug("winrm-elevated requested," \
+          " loading WinRM-elevated gem (#{spec_version})")
+        load_dependency("winrm-elevated", spec_version, "Could not load or activate WinRM-elevated")
+      end
+
+      def load_dependency(gem_name, spec_version, error_message)
+        gem gem_name, spec_version
+        first_load = require gem_name
+        load_winrm_transport!(gem_name)
 
         if first_load
-          logger.debug("WinRM::FS library loaded")
+          logger.debug("#{gem_name} library loaded")
         else
-          logger.debug("WinRM::FS previously loaded")
+          logger.debug("#{gem_name} previously loaded")
         end
       rescue LoadError => e
         logger.fatal(
-          "The `winrm-fs' gem is missing and must" \
+          "The `#{gem_name}' gem is missing and must" \
           " be installed or cannot be properly activated. Run" \
-          " `gem install winrm-fs --version '#{spec_version}'`" \
+          " `gem install #{gem_name}  --version '#{spec_version}'`" \
           " or add the following to your Gemfile if you are using Bundler:" \
-          " `gem 'winrm-fs', '#{spec_version}'`."
+          " `gem '#{gem_name}', '#{spec_version}'`."
         )
         raise Train::UserError,
-          "Could not load or activate WinRM::FS (#{e.message})"
+          "Could not load or activate #{error_message} (#{e.message})"
```

Comment:
Since `error_message` is "Could not load or activate WinRM::FS" or "Could not load or activate WinRM-elevated" this will render as "Could not load or activate Could not load or activate WinRM::FS <some error message> "

It might be worth swapping out `error_message` for `gem_name` or possibly truncating the "Could not load or activate" bit at the beginning here. 

Overall looking pretty good! :)